core/iwasm/aot/aot_runtime.c

```diff
@@ -3059,14 +3049,96 @@ aot_dump_call_stack(WASMExecEnv *exec_env)
         frame.module_offset = 0;
         frame.func_index = cur_frame->func_index;
         frame.func_offset = 0;
+        frame.func_name_wp =
+            get_func_name_from_index(module_inst, cur_frame->func_index);
 
         if (!bh_vector_append(module_inst->frames.ptr, &frame)) {
             bh_vector_destroy(module_inst->frames.ptr);
-            return;
+            return false;
         }
 
         cur_frame = cur_frame->prev_frame;
     }
+
+    return true;
+}
+
+#define PRINT_OR_DUMP()                                                \
+    do {                                                               \
+        if (print) {                                                   \
+            os_printf("%s", line_buf);                                 \
+        }                                                              \
+        else if (buf) {                                                \
+            uint32 remain_len = len - total_len;                       \
+            uint32 string_len =                                        \
+                snprintf(buf + total_len, remain_len, "%s", line_buf); \
+            if (string_len >= remain_len) {                            \
+                /* Buffer full */                                      \
+                return len;                                            \
+            }                                                          \
+            total_len += string_len;                                   \
+        }                                                              \
+        else {                                                         \
+            total_len += strlen(line_buf);                             \
+        }                                                              \
+    } while (0)
```

Comment:
The macro is called three times, how about changing it to a function to reduce the footprint? e.g.
```
static uint32
dump_call_stack(const char *line_buf, bool dump_or_print, char *buf, uint32 len)

buf is the place to output the line_buf, not the beginning of total buffer
len is the left size of the buf, not the total size of the buf
return value: the size of buffer printed
```

Reply:
`dump_call_stack` should have the same implementation in both `aot_runtime.c` and `wasm_runtime.c`

how about put it into wasm_runtime_common and rename to `wasm_runtime_dump_call_stack_impl`?

Reply:
The function name is a little long, how about `wasm_runtime_dump_line_buf`?

Reply:
Sure, this function will not be exported to developer, we can add some suffix to distinguish, e.g. `wasm_runtime_dump_line_buf_impl`